src/components/views/dialogs/DevtoolsDialog.js

```diff
@@ -605,12 +614,98 @@ class ServersInRoomList extends React.PureComponent {
     }
 }
 
+const PHASE_MAP = {
+    [PHASE_UNSENT]: "unsent",
+    [PHASE_REQUESTED]: "requested",
+    [PHASE_READY]: "ready",
+    [PHASE_DONE]: "done",
+    [PHASE_STARTED]: "started",
+    [PHASE_CANCELLED]: "cancelled",
+};
+
+function VerificationRequest({txnId, request}) {
+    const [, updateState] = useState();
+    const [timeout, setRequestTimeout] = useState(request.timeout);
+
+    /* Re-render if something changes state */
+    useEffect(() => {
+        request.on("change", updateState);
+        return () => request.off("change", updateState);
+    }, [request]);
```

Comment:
can you use `useEventEmitter(request, "change", updateState);` as Encryption Panel does